### PR TITLE
OCD-4333 - Add Standards attribute to criteria

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationCriterionAttributeDAO.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public List<CertificationCriterion> getCriteriaForTestTools() {
 
     public List<CertificationCriterion> getCriteriaForFunctionalitiesTested() {
         return getAllCriteriaAttributeEntities().stream()
-                .filter(att -> att.getFunctionalityTested())
+                .filter(att -> att.getStandard())
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationCriterionAttributeDAO.java:38]: Should this be in a different function? Something looks odd here

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/CertificationResult.java [hunk 2]
```diff
@@ -334,6 +335,16 @@ public class CertificationResult implements Serializable {
     @Builder.Default
     private List<CertificationResultSvap> svaps = new ArrayList<CertificationResultSvap>();
 
+    // TODO - Need this text for OCD-4333
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/CertificationResult.java:348]: Either we need this text, or the TODO should be removed

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/xmlgenerator/CertificationResultStandardXmlGenerator.java
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/xmlgenerator/CertificationResultStandardXmlGenerator.java:1]: Checkstyle issue

**Review by andlar — COMMENTED** [review]
I don't know where it should be, but do we need to add something to the upload parser to put the expected column heading for the new "STANDARD" field that our upload template will use? It could easily be here already, but I don't know where to look

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/CertificationResult.java (hunk 2 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/CertificationResult.java:348]: A checklist item exists xsd/json documentation exists on the ticket

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/ListingConfirmationManager.java [hunk 3]
```diff
@@ -237,6 +240,7 @@ private void saveCertificationResult(Long listingId, CertificationResult certRes
             saveTestData(certResult);
             saveTestProcedures(certResult);
             saveFunctionalityTested(certResult);
+            saveStandard(certResult);
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/ListingConfirmationManager.java]: Should be plural

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/standard/StandardDAO.java [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+                    mapping.put(standardCriterion.getId(), new ArrayList<Standard>());
+                }
+                mapping.get(standardCriterion.getId()).add(standard);
+            });
+    }
+
+    private List<StandardEntity> getAllEntities() {
+        return entityManager
+                .createQuery("SELECT DISTINCT s "
+                            + "FROM StandardEntity s "
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/standard/StandardDAO.java:187]: Need to LEFT JOIN FETCH s.rule. Check below methods with HQL as well.

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**